### DISCUSSION TITLE
How to make a field read only ?

### DISCUSSION THREAD

**Discussion by MahmudulHassan5809** [fastapi/fastapi#6243] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
class DepositBase(BaseModel):
    type: DepositTypeEnum
    cheque_no: str
    bank_name: str 
    issue_date: date
    attachment_url: str
    account_id: str
    amount: float
    attachment_url: str

    class Config:
        orm_mode = True


class DepositUpdate(DepositBase):
    type: Optional[str]
    amount: Optional[float] 
    status: Optional[str]
    attachment_url: Optional[str]
    account_id: Optional[str]
```

### Description

i have to pydantic model. One is DepositBase and another one DepositUpdate. In deposit update to i want to make a field read only from parent class.

I want to exclude account_id from request body while updating. This field should  read only. how can i do this 

### Operating System

Linux

### FastAPI Version

0.78.0

### Python Version

3.8

**Comment by zoliknemet** [comment, marked as answer]
If I understood correctly your model I would do it like this:

```python
class DepositTypeEnum(str, Enum):
    TYPE_A = "TYPE_A"
    TYPE_B = "TYPE_B"


class DepositBase(BaseModel):
    type: DepositTypeEnum
    cheque_no: str
    bank_name: str
    issue_date: date
    amount: float
    attachment_url: str


class DepositDb(DepositBase):
    account_id: str

    class Config:
        orm_mode = True


class DepositResponse(DepositDb):
    some_extra_field_only_for_response: int  # could be some aggregation


class DepositUpdate(DepositBase):
    status: Optional[str]

```

`DepositBase` has everything that your Deposit request, response and database models have in common. `DepositeDb` has everything needed for database, `DepositeUpdate` has everything you need on update (account_id is excluded and it can not be set), `DepositeResponse` can have everything from database with some extra fields if you need them

I still do not understand this `status` i `DepositeUpdate` - maybe it is suppose to be in `DepositeBase` but i do not have your logic, only models
